ib_cni_test.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+export WORKSPACE=${WORKSPACE:-/tmp/k8s_$$}
+export LOGDIR=$WORKSPACE/logs
+export ARTIFACTS=$WORKSPACE/artifacts
+
+export GOROOT=${GOROOT:-/usr/local/go}
+export GOPATH=${WORKSPACE}
+export PATH=/usr/local/go/bin/:$GOPATH/src/k8s.io/kubernetes/third_party/etcd:$PATH
+export TIMEOUT=${TIMEOUT:-300}
+
+export POLL_INTERVAL=${POLL_INTERVAL:-10}
+export NETWORK=${NETWORK:-'192.168'}
+
+export KUBECONFIG=${KUBECONFIG:-/var/run/kubernetes/admin.kubeconfig}
+pushd $WORKSPACE
+
+
+function pod_create {
+    cd $ARTIFACTS
+    wget https://raw.githubusercontent.com/Mellanox/k8s-rdma-shared-dev-plugin/master/example/test-hca-pod.yaml -O $ARTIFACTS/test-hca-pod.yaml
+    patch <<EOF
+--- test-hca-pod.yaml	2020-03-16 19:43:32.002735348 +0200
++++ test-hca-pod.yaml	2020-03-16 19:37:05.122828220 +0200
+@@ -2,6 +2,8 @@ apiVersion: v1
+ kind: Pod
+ metadata:
+   name: mofed-test-pod
++  annotations:
++    k8s.v1.cni.cncf.io/networks: ipoib-network
+ spec:
+   restartPolicy: OnFailure
+   containers:
+EOF
+    kubectl create -f $ARTIFACTS/test-hca-pod.yaml
+
+    POD_NAME=$(cat $ARTIFACTS/test-hca-pod.yaml|grep name|head -n1|awk '{print $NF}')
+
+    pod_status=$(kubectl get pods | grep $POD_NAME |awk '{print $3}')
+    let stop=$(date '+%s')+$TIMEOUT
+    d=$(date '+%s')
+    while [ $d -lt $stop ]; do
+        echo "Waiting for pod to became Running"
+        pod_status=$(kubectl get pods | grep $POD_NAME |awk  '{print $3}')
+        if [ "$pod_status" == "Running" ]; then
+            return 0
+        fi
+        kubectl get pods | grep $POD_NAME
+        kubectl describe pod $POD_NAME
+        sleep ${POLL_INTERVAL}
+        d=$(date '+%s')
+    done
+    echo "Error ${POD_NAME} is not up"
+    return 1
+}
+
+
+function test_pod {
+    status=0
+    POD_NAME=$(cat $ARTIFACTS/test-hca-pod.yaml|grep name|head -n1|awk '{print $NF}')
+    kubectl exec -i $POD_NAME -- ip a
+    echo "Checking net1 for address network $NETWORK"
+    kubectl exec -i mofed-test-pod ifconfig net1
+    n=$(kubectl exec -i mofed-test-pod ifconfig net1 | egrep "192.168|infiniband"|wc -l)
+    if [ $n -ne 2 ]; then
+	status=1
+        echo "Failed to recognize net1 as infiniband interface with $NETWORK"
+    else
+        echo "Passed to recognize net1 as infiniband interface with $NETWORK"
+    fi
+    return $status
+}
+
+
+pod_create
+test_pod
```

Comment:
please extend the test to create 2 pods and run ib_write_bw. Mamduh has the commands

Reply:
done